### PR TITLE
Query filter improvements (task #6717)

### PR DESCRIPTION
**Pull request by georgeconstantinou**

### CODE REVIEW

File: src/Event/Model/ModelBeforeFindEventsListener.php [hunk 1]
```diff
@@ -35,7 +35,10 @@ class ModelBeforeFindEventsListener implements EventListenerInterface
     public function implementedEvents()
     {
         return [
-            'Model.beforeFind' => 'filterByUserCapabilities',
+            'Model.beforeFind' => [
+                'callable' => 'filterByUserCapabilities',
+                'priority' => 999999999999 // this callback should be executed as late as possible.
```
Comment (mamchenkov) [inline src/Event/Model/ModelBeforeFindEventsListener.php]: @georgeconstantinou maybe using `PHP_INT_MAX - 1`, just to ensure that we don't overflow the maximum integer number. :)

Comment (georgeconstantinou) [reply src/Event/Model/ModelBeforeFindEventsListener.php]: @mamchenkov done: https://github.com/QoboLtd/cakephp-roles-capabilities/pull/119/commits/0719a8cc5f68e215410bd9e5f8c62d43b3e9c659

File: src/FilterQuery.php [hunk 2]
```diff
@@ -84,7 +84,7 @@ public function __construct(QueryInterface $query, RepositoryInterface $table, a
         $this->table = $table;
         $this->user = $user;
 
-        if (empty($user) || $this->isSuperuser() || $this->isSkipTable()) {
+        if (empty($this->user) || $this->isSuperuser() || $this->isSkipTable()) {
```
Comment (mamchenkov) [inline src/FilterQuery.php]: @georgeconstantinou I think we can no use `$user->isAdmin`, which will check for both the super user and admin role

Comment (georgeconstantinou) [reply src/FilterQuery.php]: this is only available through our application layer

**Pull request merged**